File: python_kata/src/trees/binary_tree/node.py
from __future__ import annotations
from typing import TypeVar


T = TypeVar("T")

class BinaryNode[T]:
    val: T
    left: BinaryNode[T] | None
    right: BinaryNode[T] | None

    def __init__(self, value: T):
        self.val = value
        self.left = self.right = None

File: typescript-kata/src/trees/binary-tree/node.ts
export class BinaryNode<T> {
  val: T;
  left: BinaryNode<T> | undefined;
  right: BinaryNode<T> | undefined;

  constructor(value: T) {
    this.val = value;
    this.left = this.right = undefined;
  }
}

export class BinaryTree<T> {
  root: BinaryNode<T> | undefined;

  constructor(root: BinaryNode<T>) {
    this.root = root;
  }
}
